services/api-server/src/simcore_service_api_server/api/routes/files.py

```diff
@@ -170,6 +182,117 @@ async def upload_files(files: list[UploadFile] = FileParam(...)):
     raise NotImplementedError
 
 
+@router.post(
+    "/content",
+    response_model=FileUploadSchema,
+)
+@cancel_on_disconnect
+@catch_n_raise(
+    (NodeportsException, ValidationError, ClientError),
+    lambda e: status.HTTP_500_INTERNAL_SERVER_ERROR,
+)
+async def get_upload_links(
+    request: Request,
+    client_file: ClientFile,
+    user_id: Annotated[PositiveInt, Depends(get_current_user_id)],
+):
+    """Get upload links for uploading a file to storage
+
+    Arguments:
+        request -- Request object
+        client_file -- ClientFile object
+        user_id -- User Id
+
+    Returns:
+        FileUploadSchema
+    """
+    assert request  # nosec
+    file_meta: File = await File.create_from_client_file(
+        client_file,
+        datetime.datetime.now(datetime.timezone.utc).isoformat(),
+    )
+    _, upload_links = await get_upload_links_from_s3(
+        user_id=user_id,
+        store_name=None,
+        store_id=SIMCORE_LOCATION,
+        s3_object=parse_obj_as(
+            StorageFileID, f"api/{file_meta.id}/{file_meta.filename}"
+        ),
+        client_session=None,
+        link_type=LinkType.PRESIGNED,
+        file_size=ByteSize(client_file.filesize),
+        is_directory=False,
+    )
+    return upload_links
+
+
+@router.patch(
+    "/content",
+    response_model=File,
+)
+@cancel_on_disconnect
+@catch_n_raise(
+    (ValueError, S3TransferError, ClientError, ValidationError),
```

Comment:
here I also have some doubts:
- ValidationError why is this a 500? what are the use-cases?
- S3TransferError? is there a use-case? cause this should happen when you actually upload to S3 directly I think.
- ClientError: is that the client of api-server having issues with storage? then I think this should be a 503

Reply:
Yep, I have removed the `catch_n_raise` decorator entirely as we discussed offline yesterday